icu4c/source/tools/icuwriteuprops/icuwriteuprops.cpp

```diff
@@ -0,0 +1,207 @@
+// © 2016 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include <iostream>
+#include "toolutil.h"
+#include "uoptions.h"
+#include "cmemory.h"
+#include "charstr.h"
+#include "cstring.h"
+#include "unicode/uchar.h"
+#include "unicode/errorcode.h"
+#include "unicode/uniset.h"
+#include "unicode/putil.h"
+#include "unicode/umutablecptrie.h"
+#include "writesrc.h"
+
+U_NAMESPACE_USE
+
+/*
+ * Global - verbosity
+ */
+UBool VERBOSE = FALSE;
+UBool QUIET = FALSE;
+
+UBool haveCopyright = TRUE;
+UCPTrieType trieType = UCPTRIE_TYPE_SMALL;
+
+void handleError(const ErrorCode& status, const char* context) {
+    if (status.isFailure()) {
+        std::cerr << "Error: " << context << ": " << status.errorName() << std::endl;
+        exit(status.get());
+    }
+}
+
+void dumpBinaryProperty(UProperty uproperty, FILE* f) {
+    ErrorCode status;
+    const char* fullPropName = u_getPropertyName(uproperty, U_LONG_PROPERTY_NAME);
+    const char* shortPropName = u_getPropertyName(uproperty, U_SHORT_PROPERTY_NAME);
+    const USet* uset = u_getBinaryPropertySet(uproperty, status);
+    handleError(status, fullPropName);
+
+    fputs("[[binary_property]]\n", f);
+    fprintf(f, "long_name = \"%s\"\n", fullPropName);
+    fprintf(f, "short_name = \"%s\"\n", shortPropName);
+    usrc_writeUnicodeSet(f, uset, UPRV_TARGET_SYNTAX_TOML);
+}
+
+void dumpEnumeratedProperty(UProperty uproperty, FILE* f) {
+    ErrorCode status;
```

Comment:
```suggestion
    IcuToolErrorCode status;
```
... *or better* pass the caller's `ErrorCode &` in as a parameter, as usual.